richtextfx/src/main/java/org/fxmisc/richtext/model/Paragraph.java

```diff
@@ -132,7 +132,8 @@ public String substring(int from) {
             int segIdx = pos.getMajor();
             List<SEG> segs = new ArrayList<>(segIdx + 1);
             segs.addAll(segments.subList(0, segIdx));
-            segs.add(segmentOps.subSequence(segments.get(segIdx), 0, pos.getMinor()));
+            segmentOps.subSequence(segments.get(segIdx), 0, pos.getMinor())
+                            .ifPresent(seg -> segs.add(seg));
```

Comment:
If we wrap the returned value of subsequence in an `Optional`, it adds boilerplate to the code (and is honestly so confusing that I'm not even sure if I've implemented it correctly, nor would I want to ever have to read through this code). See [here](https://github.com/JordanMartinez/RichTextFX/commit/a6b130124de4133763e7b5756e13d9d2e4fb7418) for my work (Note: this does not compile as ReadOnlyStyledDocument would need to account for the new changes).